### PR TITLE
Bring back huggingface support

### PR DESCRIPTION
**Pull request by tengyifei**
Since `tp run` and `tp docker-run` takes an arbitrary Python script, it's possible to run Hugging Face trainers to produce a reference loss curve. This feature bit-rotted a bit and I brought it back and documented it some more.

In particular, I removed the `tensorflow` dependency that causes a protobuf conflict. It's good to remove unneeded deps to make the build faster anyways.

TEST:

```
tp run --use-hf torchprime/hf_models/train.py train_script.args.per_device_train_batch_size=256 +train_script.args.log_loss=true train_script.args.logging_strategy=steps +train_script.args.logging_steps=1 +train_script.args.logging_first_step=true +train_script.args.report_to=tensorboard train_script.args.max_steps=1000
```

Here's a link to an internal tensorboard showing two Hugging Face runs, with constant or linear decay learning rate: http://tb/share/S3Gouj8MwgJ7xMJbz6cRM.

### CODE REVIEW

File: torchprime/launcher/cli.py [hunk 1]
```diff
@@ -277,6 +277,7 @@ def docker_run(args, use_hf: bool):
     "--privileged",
     "--net",
     "host",
+    "--shm-size=16G",
```
[the remark this answers is not in the document]
Comment (tengyifei) [reply torchprime/launcher/cli.py:280]: I'm not sure what this does. But I copied it from https://github.com/pytorch/xla/blob/9871aed555dddc895ab5dfe6f4ebdd352cf65c20/.devcontainer/tpu-contributor/devcontainer.json#L4. It looks like folks previously working with PyTorch/XLA on TPU found this to be useful: https://github.com/pytorch/xla/commit/0268a5f6db4e5122709b4af27b9b0133533e0b40

File: torchprime/hf_models/README.md [hunk 2]
```diff
@@ -21,4 +25,28 @@ add flag `--use-hf` to `tp run` command:
...
+# Run Llama 3.1 405B Hugging Face distributed training on an XPK cluster.
+tp run --use-hf torchprime/hf_models/train.py \
+    train_script.args.config_name=torchprime/hf_models/configs/model/llama-3/config_405b.json \
+    train_script.args.per_device_train_batch_size=64 \
+    train_script.args.fsdp_config=null \
+    train_script.args.fsdp=null \
+    train_script.args.tokenizer_name="meta-llama/Meta-Llama-3.1-405B" \
+    +train_script.args.spmd_2d_sharding=4
+
+# Locally run the Hugging Face trainer and log metrics to tensorboard for analysis.
```
Comment (vlasenkoalexey) [inline torchprime/hf_models/README.md:41]: would this train llama3.1 8b model?

Could you please call it out.

File: torchprime/hf_models/README.md [hunk 3]
```diff
@@ -21,4 +25,28 @@ add flag `--use-hf` to `tp run` command:
...
+
+```sh
+# Run Llama 3.1 405B Hugging Face distributed training on an XPK cluster.
+tp run --use-hf torchprime/hf_models/train.py \
+    train_script.args.config_name=torchprime/hf_models/configs/model/llama-3/config_405b.json \
+    train_script.args.per_device_train_batch_size=64 \
+    train_script.args.fsdp_config=null \
+    train_script.args.fsdp=null \
+    train_script.args.tokenizer_name="meta-llama/Meta-Llama-3.1-405B" \
+    +train_script.args.spmd_2d_sharding=4
```
Comment (vlasenkoalexey) [inline torchprime/hf_models/README.md:39]: could you please add a paragraph explaining how sharding would be applied for HF model

File: torchprime/hf_models/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (tengyifei) [reply torchprime/hf_models/README.md:41]: It will write a protobuf file recording the losses and learning rates, etc.

There won't be a link unfortunately. Neither will that happen for the PyTorch/XLA trainer.
We'd have to manually launch tensorboard. I'll document this.

Comment (tengyifei) [reply torchprime/hf_models/README.md:41]: DONE

File: torchprime/hf_models/README.md (hunk 3 above)
Comment (tengyifei) [reply torchprime/hf_models/README.md:39]: Yes
